### PR TITLE
ci: add cron for compatibility tests on main.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

I believe this should do the trick. In a non workflow-dispatch event (schedule in this case), `input` context should not contain any of the fields and the default values will be picked up.

Unfortunately, don't know of a good way to test this apart from seeing what happens on Sunday :smile: 

closes: #5762

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: .github/workflows/e2e-compatibility.yaml [hunk 1]
```diff
@@ -1,5 +1,11 @@
+# Runs compatibility tests for ibc-go.
+# Can be triggered manually by setting values for release-branch and ibc-go-version.
+# On a weekly schedule with default values of 'main' for release-branch and 'main-cron-job' for ibc-go-version.
 name: Compatibility E2E
 on:
+  schedule:
+    # run on 20:00 on Sunday.
```
Comment (bznein) [inline .github/workflows/e2e-compatibility.yaml:7]: Isn't this Saturday?

File: .github/workflows/e2e-compatibility.yaml [hunk 2]
```diff
@@ -22,7 +28,8 @@ env:
   REGISTRY: ghcr.io
   ORG: cosmos
   IMAGE_NAME: ibc-go-simd
-  RELEASE_BRANCH: '${{ inputs.release-branch }}'
+  RELEASE_BRANCH: ${{ inputs.release-branch || 'main' }}
+  IBC_GO_VERSION: ${{ inputs.ibc-go-version || 'main-cron-job' }}
```
Comment (damiannolan) [inline .github/workflows/e2e-compatibility.yaml:32]: Wonder if there's anything that's gonna cry over this? could just use `latest` if it comes to whinge

**Review by damiannolan — APPROVED** [review]
Nice! I think we can refine this if we need any test exclusions or want to tweak things a little bit

File: .github/workflows/e2e-compatibility.yaml (hunk 1 above)
Comment (bznein) [reply .github/workflows/e2e-compatibility.yaml:7]: Ah I see! Super weird that they themselves link to crontab.guru which says it's Saturday :D https://crontab.guru/#0_20_*_*_6 but  thanks for checking!

**Review by chatton — APPROVED** [review]
LGTM! Nice one

**Pull request merged**